File: Segundo-Bimestre/Parcial 1/Matrices/Matrices_python/Matrices6.py
print("")
# Inicializar las variables
i=0; j=0; m = 0; n = 0; k = 0; suma = 0; selec = 0; mayor = 0; aux = 0

# Ingresar los datos
m = int(input("Ingrese el número de trabajadores: "))
k = int(input("Ingrese el número de criterios a evaluar: "))
n = int(input("Número de aspirantes a seleccionar: "))

# Declarar el vector de trabajadores
codigo = [None]*m

# Ingresar los datos del vector código
for i in range (0, m):
    codigo[i] = input("Ingrese el nombre "+str(i)+": ")

# Declarar la matriz
C = []

for i in range (0, m):
    C.append([])
    for j in range (0, k):
        valor = int(input("Ingrese la calificación del criterio "+str(j)+" de "+str(codigo[i])+": "))
        C[i].append(valor)
        if ((C[i][j]>10) or (C[i][j]<0)):
            print("****Ingrese calificaciones dentro del rango de 0 a 10 puntos*****")
            print("***EL PROCESO HA FINALIZADO***")
            print("***REINICIE EL PROGRAMA")
            print("Los datos se obtendrán hasta antes del error detectado")
            j = k
            i = m

# Declarar el arreglo de la media
Media = [None]*m

print("***MEDIA***")
for i in range (0,m):
    for j in range (0, k):
        suma = suma + C[i][j]
        Media[i] = suma/k
    print(codigo[i],": ",Media[i],"    ")
    suma = 0

# Ordenar el vector Media
for i in range (0,m):
    for j in range (i+1, k):
        if (Media[i]>Media[j]):
            mayor = Media[i]

# Seleccionar los aspirantes para n puestos
Selec = [None]*n
aux1 = 0

for i in range (0,m):
    for j in range (0, k):
        if (Media [i] > Media[j]):
            aux = Media[i]
            Media[i] = Media[j]
            Media[j] = aux
            # Ordenar tambien el vector codigo
            aux1 = codigo[i]
            codigo[i] = codigo[j]
            codigo[j] = aux1

print("***ASPIRANTES SELECCIONADOS***")
print("Para ",n," puestos de trabajo: ")
for i in range (0, n):
    print(codigo[i]," es aspirante seleccionado con una media de: ",Media[i])

File: Segundo-Bimestre/Parcial 1/Matrices/Matrices_java/src/matrices/Matrices6.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package matrices;
import java.util.Scanner;
/**
 *
 * @author DELL
 */
public class Matrices6 {
    public static void main(String[] args) {
        Scanner leer = new Scanner (System.in);
        
        // Declarar e inicializar las variables
        int i, j, m, n, k, selec;
        double suma, mayor, aux;
        m = 0; n = 0; k = 0; suma = 0; selec = 0; mayor = 0; aux = 0;
        
        // Ingresar los datos
        System.out.print("Ingrese el número de trabajadores: ");
        m = leer.nextInt();
        System.out.print("Ingrese el número de criterios a evaluar: ");
        k = leer.nextInt();
        System.out.print("Número de aspirantes a seleccionar: ");
        n = leer.nextInt();
        
        // Declarar el vector de trabajadores
        String [] codigo = new String [m];
        
        // Ingresar los datos del vector codigo
        for (i=0;i<m;i++) {
            System.out.print("Ingrese el nombre "+i+": ");
            codigo[i] = leer.next();
        }
        
        // Declarar la matriz
        int [][] C = new int [m][k];
        
        
        for (i=0;i<m;i++) {
            System.out.println("***Calificación criterios de "+codigo[i]+"***");
            for (j=0; j<k; j++) {
                System.out.print("Ingrese la calificación del criterio "+j+" de "+codigo[i]+": ");
                C[i][j] = leer.nextInt();
                if ((C[i][j]>10) || (C[i][j]<0)) {
                    System.out.println("\t****Ingrese calificaciones dentro del rango de 0 a 10 puntos*****");
                    System.out.println("\t***EL PROCESO HA FINALIZADO***");
                    System.out.println("\t****REINICIE EL PROGRAMA***");
                    System.out.println("Los datos se obtendrán hasta antes de error detectado");
                    j = k;
                    i = m;
                }
            }
        }
        
        // Declarar el arreglo de media
        double [] Media = new double [m];
        
        System.out.println("***Media***");
        for (i=0;i<m;i++) {
            for (j=0;j<k;j++) {                
                suma = suma + C[i][j];
                Media[i] = suma/k;
            }
            System.out.println(codigo[i]+": "+Media[i]+"   ");
            suma = 0;
        }
        
        // Ordenar el vector media
                
        for (i=0;i<m;i++) {
            for (j=i+1;j<k;j++) {
                if (Media[i]>Media[j]) {
                    mayor = Media[i];
                }
            }
        }        
        
        // Seleccionar los aspirantes para n puestos
        double [] Selec = new double [n];
        String aux1;
        
        for (i=0;i<m;i++) {
            for (j=0;j<k;j++) {
                if (Media [i] > Media[j]) {
                    aux = Media[i];
                    Media [i] = Media[j];
                    Media [j] = aux;
                    // Ordenar también los nombres de los aspirantes
                    aux1 = codigo[i];
                    codigo[i] = codigo[j];
                    codigo[j] = aux1;
                }
            }
        }
        
        System.out.println("***ASPIRANTES SELECCIONADOS***");
        System.out.println("Para "+n+" puestos de trabajo: ");
        for (i=0; i<n; i++) {
                System.out.println(codigo[i]+" es apirante seleccionado con una media de: "+Media[i]);
        }        
    }
}
